src/core/utils/tree.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function isSubtree(leafPath: string, rootPath: string, delimiter = '.') {
+  if (!leafPath || !rootPath) return false;
+  const leafSegments = leafPath.split(delimiter);
+  const rootSegments = rootPath.split(delimiter);
+  let i = 0;
```

Comment:
__nit:__ looks like a shorter version would be (I'm confident `every` will do an early return as well):
```ts
return rootSegments.every((rootSegment, index) => leafSegments[index] === rootSegment);
```

Reply:
yeah, it does. will change, thanks